File: Rails/app/models/affectation.rb
class Affectation < ApplicationRecord
  belongs_to :festival
  belongs_to :task
  belongs_to :user

  validates :expected_start, :expected_end, :responsability, presence: true


  validate :expected_end_after_expected_start

   private

  def expected_end_after_expected_start
    return if expected_end.blank? || expected_start.blank?

    if expected_end < expected_start
      errors.add(:expected_end, "doit etre apres la date de debut")
    end
  end


  
end

File: Angular/src/app/core/models/affectation.ts
import { Festival } from "./festival"
import { Task } from "./task"
import { User } from "./user"

export class Affectation {
    id: number
    expected_start: string
    expected_end: string
    responsability: string
    task: Task
    festival:Festival
    user:User
    start:string|null
    end:string|null
   
   
    
    constructor(
       id: number,
       expected_start: string,
       expected_end: string,
       responsability: string,
       task: Task,
       festival:Festival,
       user:User,
       start: string,
       end: string
       
    ) {
        this.id = id;
        this.task = task;
        this.festival = festival;
        this.user = user;
        this.expected_start = expected_start;
        this.expected_end = expected_end;
        this.responsability = responsability;
        this.start = start; 
        this.end = end;
    }
}